### PR TITLE
update log4j to 2.15.0

### PR DESCRIPTION
**Pull request by dapowers87**
Signed-off-by: David Powers <ddpowers@amazon.com>

### Description
update log4j to 2.15.0
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: THIRD-PARTY [hunk 1]
```diff
@@ -1,3 +1,50 @@
+** MustacheJava; version 0.9.6 -- https://github.com/spullara/mustache.java
```
Comment (dlvenable) [inline THIRD-PARTY:1]: These seem like a lot of changes for this PR. Did you re-run the following command to generate this?

```
./gradlew generateThirdPartyReport
```

File: build.gradle [hunk 2]
```diff
@@ -51,7 +51,9 @@ subprojects {
     dependencies {
         implementation 'com.google.guava:guava:31.0.1-jre'
         implementation platform('io.micrometer:micrometer-bom:1.7.5')
+        implementation 'org.apache.logging.log4j:log4j-core:2.15.0'
```
Comment (dlvenable) [inline build.gradle:54]: I do not believe that we should have `log4` in the subprojects. This should only be in Data Prepper core.

File: build.gradle [hunk 3]
```diff
@@ -51,7 +51,9 @@ subprojects {
...
         implementation 'org.slf4j:slf4j-api:1.7.32'
+        implementation 'org.apache.logging.log4j:log4j-slf4j-impl:2.15.0'
```
Comment (dlvenable) [inline build.gradle:56]: See my comment above. I believe this need only be in data-prepper-core.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: THIRD-PARTY (hunk 1 above)
Comment (dapowers87) [reply THIRD-PARTY:1]: No, I simply cherry picked and merged.

Comment (dlvenable) [reply THIRD-PARTY:1]: The `v1.1.x` branch used a different format for the `THIRD-PARTY` file. I'd say that you should remove the changes here and then run:

```
./gradlew generateThirdPartyReport
```

That command will update your `THIRD-PARTY` file. Commit that file and include it in the PR.

File: build.gradle (hunk 2 above)
Comment (dlvenable) [reply build.gradle:54]: This code block configures dependencies for all sub-projects. So any dependency here is used in all sub-projects.

Data Prepper uses SLF4J which provides a logging abstraction. All the sub-projects need SLF4J. Indeed it is included here:

```
implementation 'org.slf4j:slf4j-api:1.7.32'
```

However, only the application needs to include an actual SLF4J implementation. The application is `data-prepper-core` and it uses Log4j as the SLF4J implementation.

This is where data-prepper-core includes Log4j: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/build.gradle#L31-L32

--

Basically, lines 54 and 56 are unnecessary and should just be deleted.

**Review by dlvenable — APPROVED** [review]
